### ISSUE TITLE
ESEA Demos with missing playbackTime, ticks, frames & signonLength

### ISSUE THREAD

**Issue by mavrick** [saul/demofile#65]
Hey mate,

I've got a lot of ESEA demos that appear to be missing the end of the file.

```RangeError: Illegal offset: 0 <= 206597010 (+64) <= 206597072
    at ByteBuffer.module.exports.ByteBufferPrototype.readBytes (C:\test\node_modules\bytebuffer\dist\bytebuffer-node.js:499:23)
    at ByteBuffer.readIBytes (C:\test\node_modules\demofile\ext\bytebuffer.js:21:15)
    at DemoFile._handleDemoPacket (C:\test\node_modules\demofile\demo.js:207:31)
    at DemoFile._parseRecurse (C:\test\node_modules\demofile\demo.js:321:16)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

``demoFile.header`` returns:

```{ magic: 'HL2DEMO',
  protocol: 4,
  networkProtocol: 13626,
  serverName: 'Counter-Strike: Global Offensive',
  clientName: 'GOTV Demo',
  mapName: 'de_mirage',
  gameDirectory: 'csgo',
  playbackTime: 0,
  playbackTicks: 0,
  playbackFrames: 0,
  signonLength: 0 }
```

``sv_minupdaterate`` & ``sv_maxupdaterate`` are set to ``128`` - what is the best way to try to heal the header information?

**Comment by mavrick** [comment]
Nevermind, I was half asleep when I wrote this. I know how tickrates work, just had the dumb.

As for the header information returning 0's - this is because the buffer is not complete, correct?

**Comment by saul** [comment]
No, the header is written fully when the server finishes the demo. The reason the header isn’t filled is because the demo isn’t being flushed before the game ends (the process might be aggressively terminated).

**Comment by saul** [comment]
Can these demos even be played in the game client? I believe tick rate info is available on the svc_ServerInfo message.

**Comment by mavrick** [comment]
will take a look and report back

**Comment by saul** [comment]
Assume that worked - let me know if you have any further issues.

**Comment by lucasreppewelander** [comment]
I am experiencing the same issue.. I downloaded a GOTV demo from HLTV, I haven't tried to watch it through CSGO but I do get all information when parsing this demo except for the currentTime, as with @mavrick, my headers also states 0 on playbackTime and ticks etc, heres a paste of my `demo.header`:
```
{
  magic: 'HL2DEMO',
  protocol: 4,
  networkProtocol: 13739,
  serverName: 'Counter-Strike: Global Offensive',
  clientName: 'GOTV Demo',
  mapName: 'de_inferno',
  gameDirectory: 'csgo',
  playbackTime: 0,
  playbackTicks: 0,
  playbackFrames: 0,
  signonLength: 0 }
```

and here is my `svc_ServerInfo`:
```
CSVCMsg_ServerInfo {
  protocol: 13739,
  serverCount: 4,
  isDedicated: false,
  isHltv: true,
  cOs: 108,
  mapCrc: 2910759191,
  clientCrc: 1629019561,
  stringTableCrc: 0,
  maxClients: 64,
  maxClasses: 282,
  playerSlot: 0,
  tickInterval: 0.0078125,
  gameDir: 'csgo',
  mapName: 'de_inferno',
  mapGroupName: '',
  skyName: 'nukeblank',
  hostName: 'Counter-Strike: Global Offensive',
  isRedirectingToProxyRelay: false,
  ugcMapId: Long { low: 0, high: 0, unsigned: true } }
```

when I try to use the `demo.currentTime` all I get is `NaN` and when trying to do `typeof demo.currentTime` it returns undefined

any idea? this is the demo I'm currently using: https://www.hltv.org/download/demo/55348

**Comment by saul** [comment]
Fixed in v1.2.0
